# Graded assignment 1

Rick Huijbregts 2014965


# Packages

We need the package `tidyverse`

In [8]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [9]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1          V2         V3        V4        V5        V6        V7       
1 0.009712653 0.98159437 0.1353513 0.1350901 0.3157167 0.1182228 0.9884597
2 0.298175013 0.02939497 0.7045813 0.3507374 0.4080129 0.1881615 0.6188040
  V8        V9        V10       V11       V12       V13       V14      
1 0.9223494 0.8495822 0.5892562 0.8040336 0.2684660 0.1083805 0.7678201
2 0.8955856 0.9840215 0.2232802 0.2212385 0.9356605 0.9777398 0.2336949
  V15       V16       V17       V18       V19       V20      
1 0.6922612 0.3482956 0.2140115 0.9296423 0.6552616 0.7331527
2 0.2217589 0.9274042 0.6817523 0.5115920 0.7835090 0.3605095

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [10]:
df2 <- df1 %>%
gather (key, Value, V1:V20)
  

head(df2)

key Value      
1 V1  0.009712653
2 V1  0.298175013
3 V2  0.981594369
4 V2  0.029394975
5 V3  0.135351297
6 V3  0.704581287

## Assignment 2

We will generate some data:

In [11]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [12]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [13]:
df5 <- full_join(df3, df4, by = 'id')
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook